### ISSUE TITLE
Move/create authorization classes in .Net Core

### ISSUE THREAD

**Issue by pi3k14** [dotnet/corefx#28846]
The Microsoft.AspNetCore.Authorization namespace contains classes that are useful in .Net Core and is not dependent on ASP.Net Core, please make them available in a .Net Core namespace. Ref. [Make Authorization available outside of ASP.NET Core](https://github.com/aspnet/Security/issues/1480)

**Comment by karelz** [comment]
If you have an API concept in mind, please describe the concept for us who don't know anything about Microsoft.AspNetCore.Authorization.
If you just need the API surface shipped under different namespace, then it should happen where the code lives to avoid code duplication.

**Comment by pi3k14** [comment]
> If you just need the API surface shipped under different namespace, then it should happen where the code lives to avoid code duplication.

Yes, and it should live in .Net Core, because there is nothing ASP.Net specific about it. 
The ASP.Net specific parts are in the Microsoft.AspNetCore.Authorization.Policy namespace (some concepts there are also more generic).

There is a common concept her between .Net Framework, ASP.Net Core and .Net Core.
Eg. in .Net Framework you would implement the Policy check with the help of  CodeAccessSecurityAttribute. How to use this concept in ASP.Net Core is described [here](https://docs.microsoft.com/en-us/aspnet/core/security/authorization/policies).
Just to call an evaluator method, instead of theese attributes, would be sufficient. Something that would be excellent in an DDD solution, where you do this check in the Application Layer.

Including an ASP.Net namespace/assembly into something that is not ASP at all is something you wouldn't do, because you have no idea what will break further down the road.

Documentation of the ASP.Net Core library you can find [here](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.authorization.authorizationpolicy?view=aspnetcore-2.0).
